### ISSUE TITLE
Glide startup can lock up due in unmarshalling depending on network

### ISSUE THREAD

**Issue by prashantv** [Masterminds/glide#329]
If I had a glide.yaml with the following import
```
import:
- package: some.internal.domain/team/project
```

If I run `glide list`, it will deserialize this YAML, and as part of that step, `Dependency.UnmarshalYAML` calls `util.NormalizeName(d.Name)` which calls `GetRootFromPackage(name)`. This ends up trying to hit the network, and can cause a long stall depending on which network I'm connected to (~30 seconds) with no log output or any idea of what is happening.

I think:
1) Unmarshalling the `glide.yaml` should not be relying on network behaviour. Especially not for something like `glide list`. The code should be moved out of the unmarshal path
2) There should be some sort of logging when making a call that can fail

**Comment by technosophos** [comment]
Yeah, I traced this down to `getRootFromGoGet`, which calls an `http.Get`.

The reason for this is because certain packages uses aliases built into HTTP headers. We might be able to special-case this for `glide list`... but we'd lose some functionality.

I wonder how feasible it is to shorted the timeout to 5 seconds instead of 30. Suggestions, @mattfarina ?

**Comment by prashantv** [comment]
A shorter timeout would be great. I think there should be some logging (or at least debug logging) when it tries to hit any URL for others trying to debug why it's taking so long.
